src-self-hosted/arg.zig

```diff
@@ -0,0 +1,284 @@
+const std = @import("std");
```

Comment:
I haven't put this argument parsing code into `std` since I haven't really written it with the general case in mind and there are probably other things we would like to support and consider.